src/Symfony/Component/Routing/Loader/Configurator/Traits/RouteTrait.php

```diff
@@ -124,4 +124,18 @@
 
         return $this;
     }
+
+    /**
+     * Add a prefix to the name of all the routes within the collection.
+     *
+     * @param string $prefix
+     *
+     * @return $this
+     */
+    final public function addNamePrefix(string $prefix)
```

Comment:
`: self` to fully get rid of the phpdoc?

Reply:
Well, we still need the description I'd say, as it's meant to be used by developers

Reply:
Getting rid of `@return $this` would be nice though.

Reply:
but... on the trait I'm not sure if this makes sense... as (at least in my mind) `self` refers to the trait here, not the class `use`ing the trait.

Reply:
There is no such thing as referring to a trait. PHP merges traits into the class, so self will always point to the correct class.

Reply:
I was talking conceptually, by reading this. But if you two believe it's better just to put `self` within the trait, I'm happy with it as I don't mind that much tbh. Changed 👍